homeassistant/components/knx/notify.py

```diff
@@ -7,70 +7,50 @@
 from homeassistant.core import HomeAssistant
-from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType
 
 from .const import DATA_KNX_CONFIG, DOMAIN, KNX_ADDRESS
-from .schema import NotifySchema
+from .knx_entity import KnxEntity
 
 
-async def async_get_service(
+async def async_setup_entry(
     hass: HomeAssistant,
-    config: ConfigType,
-    discovery_info: DiscoveryInfoType | None = None,
-) -> KNXNotificationService | None:
-    """Get the KNX notification service."""
-    if discovery_info is None:
-        return None
+    config_entry: config_entries.ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up notify(s) for KNX platform."""
+    xknx: XKNX = hass.data[DOMAIN].xknx
+    config: list[ConfigType] = hass.data[DATA_KNX_CONFIG][Platform.NOTIFY]
 
-    if platform_config := hass.data[DATA_KNX_CONFIG].get(NotifySchema.PLATFORM):
-        xknx: XKNX = hass.data[DOMAIN].xknx
+    async_add_entities(KNXNotify(xknx, entity_config) for entity_config in config)
 
-        notification_devices = [
-            XknxNotification(
-                xknx,
-                name=device_config[CONF_NAME],
-                group_address=device_config[KNX_ADDRESS],
-                value_type=device_config[CONF_TYPE],
-            )
-            for device_config in platform_config
-        ]
-        return KNXNotificationService(notification_devices)
 
-    return None
+def _create_notification_instance(xknx: XKNX, config: ConfigType) -> XknxNotification:
+    """Return a KNX Notification to be used within XKNX."""
+    return XknxNotification(
+        xknx,
+        name=config[CONF_NAME],
+        group_address=config[KNX_ADDRESS],
+        value_type=config[CONF_TYPE],
+    )
 
 
-class KNXNotificationService(BaseNotificationService):
-    """Implement demo notification service."""
+class KNXNotify(NotifyEntity, KnxEntity):
+    """Representation of a KNX notification entity."""
 
-    def __init__(self, devices: list[XknxNotification]) -> None:
-        """Initialize the service."""
-        self.devices = devices
+    _device: XknxNotification
 
-    @property
-    def targets(self) -> dict[str, str]:
-        """Return a dictionary of registered targets."""
-        ret = {}
-        for device in self.devices:
-            ret[device.name] = device.name
-        return ret
+    def __init__(self, xknx: XKNX, config: ConfigType) -> None:
+        """Initialize a KNX notification."""
+        super().__init__(_create_notification_instance(xknx, config))
+        self._attr_entity_category = config.get(CONF_ENTITY_CATEGORY)
+        self._attr_unique_id = str(self._device.remote_value.group_address)
 
     async def async_send_message(self, message: str = "", **kwargs: Any) -> None:
         """Send a notification to knx bus."""
```

Comment:
The service signature for the entity is slightly different.

```suggestion
    async def async_send_message(self, message: str) -> None:
        """Send a notification to knx bus."""
```